esx_service/vsan_policy.py

```diff
@@ -43,6 +44,87 @@ def create(name, content):
     return create_policy_file(filename, content)
 
 
+def update(name, content):
+    """
+    Update the content of an existing VSAN policy in the VSAN datastore.
+    Update the policy content in each VSAN object currently using the policy. If
+    a VSAN policy of the given name does not exist return an error string.
+    Return None on success.
+    """
+    path = policy_path(name)
+    if not path:
+        return ERROR_NO_VSAN_DATASTORE
+
+    err = update_policy_file_content(path, content)
+    if err:
+        return err
+
+    return update_vsan_objects_with_policy(name, content)
+
+
+def update_policy_file_content(path, content):
+    """
+    Update the VSAN policy file content. Return an error msg or None on success.
+    """
+    try:
+        with open(path) as f:
+            existing_content = f.read()
+    except OSError as e:
+        if not os.path.isfile(path):
+            return 'Error: Policy does not exist'
+        else:
+            return 'Error opening existing policy file: {0}'.format(e)
+
+    if existing_content.strip() == content.strip():
+            return 'Error: New policy is identical to old policy. Ignoring.'
+
+    # Create a temporary file so we don't corrupt an existing policy file
+    tmpfile = '{0}.tmp'.format(path)
+    err = create_policy_file(tmpfile, content)
+    if err:
+        return err
+
+    # Do an atomic rename of the tmpfile to the real policy file
+    try:
+        os.rename(tmpfile, path)
```

Comment:
Hmmm, That's a great point.  There needs to be some form of 2 phase commit here :D. Maintaining the original file as a backup should be done, and then replacing the new one with the old one if we can't update the volume. 

We can't just not do the rename first, because we actually use the policy file via objtool to update the VMs. Although we could just use a copy. Either way I'll fix it.

This brings up another problem though. The code currently allows updating policies with no existing volumes. We won't be able to do the check in that case. Ideally we'd have a way to check outside of updating/creating a volume.